ccp/lib/processors/pipeline_information/builds.py

```diff
@@ -415,33 +428,34 @@ def get_build_logs(
         :type ordered_job_list Union[list, str]
         :param build_number: The id of the build
         :type build_number str
-        :return: A list of dicts where each dict contains stage name, and stage
-        logs, as returned by get_stage_logs
-        stageflownodes.
-        None is returned on failure
+        :return: An OrderedDict of dicts where each dict contains stage name,
+        and stage logs, as returned by get_stage_logs. None is returned on
+        failure
+        :rtype OrderedDict
         """
         # TODO : Make this testable
         result = None
         stage_count = self.get_stage_count(
             ordered_job_list=ordered_job_list, build_number=build_number
         )
         if stage_count:
-            result = []
+            result = OrderedDict()
             for i in range(1, stage_count):
```

Comment:
@mohammedzee1000 : Say if a build has 10 stages, then stage_count = 10
```
In [9]: range(1, 10)
Out[9]: [1, 2, 3, 4, 5, 6, 7, 8, 9]

In [10]: len(range(1, 10))
Out[10]: 9
```

shouldn't that miss the last stage ?

Reply:
Hmm, updating to stage_count+1